tools/issue-labeler/backfill.go

```diff
@@ -0,0 +1,104 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"sort"
+
+	"github.com/golang/glog"
+)
+
+type Issue struct {
+	Number int
+	Body   string
+	Labels []Label
+}
+
+type Label struct {
+	Name string
+}
+
+type IssueUpdate struct {
+	Labels []string `json:"labels"`
+}
+
+func backfill(since string, enrolledTeams map[string][]string, dryRun bool) {
+	client := &http.Client{}
+	done := false
+	page := 1
+	var issues []Issue
+	for !done {
+		url := fmt.Sprintf("https://api.github.com/repos/hashicorp/terraform-provider-google/issues?since=%s&per_page=100&page=%d", since, page)
+		req, err := http.NewRequest("GET", url, nil)
+		if err != nil {
+			glog.Exitf("Error creating request: %v", err)
+		}
+		req.Header.Add("Accept", "application/vnd.github+json")
+		req.Header.Add("Authorization", "Bearer "+os.Getenv("GITHUB_TOKEN"))
+		req.Header.Add("X-GitHub-Api-Version", "2022-11-28")
+		resp, err := client.Do(req)
+		if err != nil {
+			glog.Exitf("Error listing issues: %v", err)
+		}
+		defer resp.Body.Close()
+		body, err := io.ReadAll(resp.Body)
+		if err != nil {
+			glog.Exitf("Error reading response body: %v", err)
+		}
+		var newIssues []Issue
+		json.Unmarshal(body, &newIssues)
+		if len(newIssues) == 0 {
+			done = true
+		} else {
+			issues = append(issues, newIssues...)
+			page++
+		}
+	}
+	for _, issue := range issues {
+		desired := make(map[string]struct{})
+		for _, existing := range issue.Labels {
+			desired[existing.Name] = struct{}{}
+		}
+		oldLength := len(desired)
+
+		for _, needed := range labels(issue.Body, enrolledTeams, false) {
+			desired[needed] = struct{}{}
+		}
+
+		if len(desired) > oldLength {
+			desiredSlice := []string{"forward/review"}
+			for label := range desired {
+				desiredSlice = append(desiredSlice, label)
+			}
+			sort.Strings(desiredSlice)
+			url := fmt.Sprintf("https://api.github.com/repos/hashicorp/terraform-provider-google/issues/%d", issue.Number)
+			update := IssueUpdate{Labels: desiredSlice}
+			body, err := json.Marshal(update)
+			if err != nil {
+				glog.Exitf("Error marshalling json: %v", err)
```

Comment:
I'm not sure we want to exit here, because we may have already started patching some of the issues, and we may be able to patch more. I believe the implementation of this command is idempotent, so we should be able to run the same job again if there are failures to retry them, and it would probably be best if the whole process doesn't stop on one failure.

Same below for the other exits.

Reply:
I will make the exits log errors and continue.